### ISSUE TITLE
ASTVisitor::visitClass() should accept ASTEnum class or an relevant interface.

### ISSUE THREAD

**Issue by lisfox1** [pdepend/pdepend#623]
- PHP Depend version: 2.11.0/master
- PHP Version: 8.1.9
- Installation type: composer
- Operating System / Distribution & Version: macOS Monterey Version 12.1

### Current Behavior
When calling `PDepend\Source\AST\ASTEnum::accept` it returns a TypeError.
`TypeError : PDepend\Source\ASTVisitor\AbstractASTVisitor::visitClass(): Argument #1 ($class) must be of type PDepend\Source\AST\ASTClass, PDepend\Source\AST\ASTEnum given`

### Expected Behavior
N/A

### Steps To Reproduce:
```
$class = new \PDepend\Source\AST\ASTEnum('test', new \PDepend\Source\AST\ASTScalarType('string'));
$class->accept(new \PDepend\Metrics\Analyzer\CodeRankAnalyzer\PropertyStrategy());
```

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/pdepend/pdepend/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/pdepend/pdepend/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "Default value is not available in AST when it's an array".

**Comment by kylekatarnls** [comment]
Hello, sorry for that, I'm building a patch, it can be tried with `composer require pdepend/pdepend:2.11.x-dev`

**Comment by lisfox1** [comment]
Hi,
Thank you for that! I can confirm that enums are evaluated correctly with the 2.11.x-dev branch

**Comment by tvbeek** [comment]
@lisfox1 thanks for your feedback

I close this issue vecause this is now fixed with version 2.11.1
